src/box/lua/net_box.lua

```diff
@@ -1201,8 +1249,12 @@ index_metatable = function(remote)
         if opts and opts.buffer then
             error("index:count() doesn't support `buffer` argument")
         end
-        local code = string.format('box.space.%s.index.%s:count',
-                                   self.space.name, self.name)
+        local code = 'box.space' .. (self.space.name ~= nil and
+                                     '.' .. self.space.name or
+                                     '[' .. self.space.id .. ']') ..
+                     '.index' .. (self.name ~= nil and
+                                 '.' .. self.name or '[' .. self.id .. ']') ..
+                     ':count'
```

Comment:
I don't know that we should do here...

```
tarantool> c.space.abc.index.bcd:count()
---
- error: Procedure 'box.space.abc.index.bcd:count' is not defined
...

tarantool> c.space['is it possible'].index.bcd:count()
---
- error: Procedure 'box.space.is it possible.index.bcd:count' is not defined
...
```

In case if space/index contains spaces we anyway fail... (It's wierd but it's possible).
And probably error message talks about function... Not about space/index existance

Reply:
AFAIC, you cannot use non-printable characters in Tarantool identifiers.

Reply:
```
tarantool> box.schema.space.create('you can')
---
- engine: memtx
  before_replace: 'function: 0x0102e5bf10'
  field_count: 0
  is_sync: false
  on_replace: 'function: 0x0102e5bed8'
  temporary: false
  index: []
  is_local: false
  enabled: false
  name: you can
  id: 517
- created
...

tarantool> box.space['you can']:create_index('do it')
---
- unique: true
  parts:
  - type: unsigned
    is_nullable: false
    fieldno: 1
  hint: true
  id: 0
  type: TREE
  space_id: 517
  name: do it
...

tarantool> box.space['you can'].index['do it']:select()
2023-04-29 11:10:00.462 [39623] main/103/interactive C> Potentially long select from space 'you can' (517)
 stack traceback:
	builtin/box/schema.lua:2497: in function 'log_long_select'
	builtin/box/schema.lua:2515: in function <builtin/box/schema.lua:2500>
	[C]: in function 'pcall'
	builtin/box/console.lua:427: in function 'eval'
	builtin/box/console.lua:819: in function 'repl'
	builtin/box/console.lua:870: in function <builtin/box/console.lua:858>
---
- []
...
```

Reply:
```console
tarantool> box.schema.space.create(' s\n')
---
- error: 'Invalid identifier '' s

    '' (expected printable symbols only or it is too long)'
...
```

I thought all whitespace symbols are considered non-printable.

Okay, well, I guess we indefinitely using square brackets could solve this.

Reply:
Fixed, added a test case for this.